lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php

```diff
@@ -3306,14 +3306,17 @@ public function setSequenceGeneratorDefinition(array $definition)
             $definition['quoted']       = true;
         }
 
-        if (! isset($definition['allocationSize']) || trim($definition['allocationSize']) === '') {
+        if (! isset($definition['allocationSize']) || trim((string) $definition['allocationSize']) === '') {
             $definition['allocationSize'] = '1';
         }
 
-        if (! isset($definition['initialValue']) || trim($definition['initialValue']) === '') {
+        if (! isset($definition['initialValue']) || trim((string) $definition['initialValue']) === '') {
             $definition['initialValue'] = '1';
         }
 
+        $definition['allocationSize'] = (string) $definition['allocationSize'];
+        $definition['initialValue']   = (string) $definition['initialValue'];
+
```

Comment:
It would make sense to use a variable here now so you need to cast only one time. It increases readability  after the type casts were introduced or when changes are going to be introduced where the type casts are forgotten.

Reply:
Can you please show me how you would do it? It tried, and it's a bit too tricky IMO, since `$definition['allocationSize']` might not be defined.

Reply:
My bad, I confused keys during my review.

Reply:
No worries :)